worlds/kh2/Rules.py

```diff
@@ -1,96 +1,1839 @@
+            if self.world.multiworld.FinalXemnas[self.player]:
+                self.world.multiworld.completion_condition[self.player] = lambda state: state.has(ItemName.Victory, self.player, 1)
+            else:
+                self.world.multiworld.completion_condition[self.player] = lambda state: state.has(ItemName.LuckyEmblem, self.player, self.world.multiworld.LuckyEmblemsRequired[self.player].value)
+        # hitlist if == 2
+        elif self.world.multiworld.Goal[self.player] == "hitlist":
+            add_rule(
+                    self.world.multiworld.get_location(LocationName.FinalXemnas, self.player),
+                    lambda state: state.has(ItemName.Bounty, self.player, self.world.multiworld.BountyRequired[self.player].value))
+            if self.world.multiworld.FinalXemnas[self.player]:
+                self.world.multiworld.completion_condition[self.player] = lambda state: state.has(ItemName.Victory, self.player, 1)
+            else:
+                self.world.multiworld.completion_condition[self.player] = lambda state: state.has(ItemName.Bounty, self.player, self.world.multiworld.BountyRequired[self.player].value)
         else:
-            world.completion_condition[player] = lambda state: state.kh_lucky_emblem_unlocked(player, world.LuckyEmblemsRequired[player].value)
-    # hitlist if == 2
-    else:
-        add_rule(world.get_location(LocationName.FinalXemnas, player),
-                 lambda state: state.kh_hitlist(player, world.BountyRequired[player].value))
-        if world.FinalXemnas[player]:
-            world.completion_condition[player] = lambda state: state.kh_victory(player)
+            add_rule(
+                    self.world.multiworld.get_location(LocationName.FinalXemnas, self.player),
+                    lambda state: state.has(ItemName.Bounty, self.player, self.world.multiworld.BountyRequired[self.player].value) and
+                                  state.has(ItemName.LuckyEmblem, self.player, self.world.multiworld.LuckyEmblemsRequired[self.player].value))
+
+            if self.world.multiworld.FinalXemnas[self.player]:
+                self.world.multiworld.completion_condition[self.player] = lambda state: state.has(ItemName.Victory, self.player, 1)
+            else:
+                self.world.multiworld.completion_condition[self.player] = lambda state: state.has(ItemName.Bounty, self.player, self.world.multiworld.BountyRequired[self.player].value) and \
+                                                                                        state.has(ItemName.LuckyEmblem, self.player, self.world.multiworld.LuckyEmblemsRequired[self.player].value)
+
+    def kh2_set_count_sum(self, item_name_set: set, state: CollectionState) -> int:
+        """
+        Returns the sum of state.count() for each item in the set.
+        """
+        return sum(
```

Comment:
why are you converting the list into a set? the if check here also does nothing

Reply:
I remember it breaking somthing when I wasnt using a set but now its working fine when I just make them a list so idk